library/src/ui/index.js

```diff
@@ -1 +1,72 @@
-window.CoNDeT.ui = {}
\ No newline at end of file
+window.CoNDeT.ui = {
+    BaseComponent: {
+        state: {
+
+        },
+        common: {
+
+        },
+        init: function(common, props) {
+            this.ref = this.onInit();
+        },
+        update: function(common, props) {
+            children.update();
+            this.onUpdate();
+            this.ref.onUpdate();
+        },
+        destroy: function(common, props) {
+            children.destroy();
+            this.onDestroy();
+            this.ref.onDestroy();
+        },
+        getPosition: function() {
+            return {
+                x: this.domReference.offsetLeft,
+                y: this.domReference.offsetTop
+            };
+        },
+        getDimensions: function() {
+            return {
+                width: this.domReference.clientWidth,
+                height: this.domReference.clientHeight
+            }
+        },
+        containsPoint: function(x, y) {
+            return (x >= this.domReference.offsetLeft && 
+                x <= this.domReference.offsetLeft + this.domReference.clientWidth &&
+                y >= this.domReference.offsetTop &&
+                y <= this.domReference.offsetTop + this.domReference.clientHeight) ? true : false;
+        },
+        setState: function(state) {
+            this.state.onDestroy();
+            this.state = state;
+            this.state.onInit();
+        },
+        setupEventListeners: function() {
+            var self = this;
+            
+            ref.addEventListener('keyup', function() {
+                self.strategy.onKeyUp();
+            });
+            ref.addEventListener('keydown', function() {
+                self.strategy.onKeyDown();
+            });
+            ref.addEventListener('mouseup', function(component) {
+                component.onInit();
+                self.strategy.onMouseUp();
+            });
+            ref.addEventListener('mousedown', function(component) {
+                component.onInit();
+                self.strategy.onMouseDown();
+            });
+        },
+        appendChild: function(child, props, position = 0) {
+            var child = new BaseComponent.init(this.common, props); // ???
+            this.children.splice(position, 0, child);
+        },
```

Comment:
Tutaj jak powiedziałem. W dokumentacji poprawię na dwie metody. `appendChild` która będzie przyjmować już gotowy komponent oraz `createChild` która dostanie `props` i referencję do kontruktora komponentu. W obu będzie opcjonalny parametr position.  
Funkcja `createChild` będzie działać tak samo jak `appendChild`, ale utworzy komponent (referencja do konstruktora) oraz wywoła init podając `common` i `props`.  
Poza dodaniem do tablicy `this.children` elementu należy zwalidować, czy tablica istnieje oraz czy podana pozycja nie wykracza poza dopuszczalny zasięg. Jeżeli wykracza, to wstawić na ostatniej pozycji.  
Należy jeszcze dodać `child.ref` w odpowiednie miejsce `this.ref`. Mając tablicę wszystkich komponentów powinno się dać użyć [tego](https://developer.mozilla.org/en-US/docs/Web/API/Node/insertBefore) biorąc `ref` następnego komponentu na liście. Jak jest wstawiany jako ostatni to [`appendChild`](https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild)